src/tcop/tcop.cpp

```diff
@@ -81,28 +163,99 @@ Result TrafficCop::ExecuteStatement(
     std::shared_ptr<stats::QueryMetric::QueryParams> param_stats,
     const std::vector<int> &result_format, std::vector<ResultType> &result,
     int &rows_changed, UNUSED_ATTRIBUTE std::string &error_message) {
+
   if (FLAGS_stats_mode != STATS_TYPE_INVALID) {
     stats::BackendStatsContext::GetInstance()->InitQueryMetric(statement,
                                                                param_stats);
   }
-
   LOG_TRACE("Execute Statement of name: %s",
             statement->GetStatementName().c_str());
   LOG_TRACE("Execute Statement of query: %s",
             statement->GetStatementName().c_str());
   try {
-    bridge::PlanExecutor::PrintPlan(statement->GetPlanTree().get(), "Plan");
-    bridge::peloton_status status = bridge::PlanExecutor::ExecutePlan(
-        statement->GetPlanTree().get(), params, result, result_format);
-    LOG_TRACE("Statement executed. Result: %d", status.m_result);
-    rows_changed = status.m_processed;
-    return status.m_result;
-  } catch (Exception &e) {
+    auto query_str = boost::to_upper_copy<std::string>(statement->GetQueryString());
+    if (query_str == "BEGIN")
```

Comment:
I think we should avoid string matching here. There should be a way to get the state of the transaction from the statement object. Also I think "ABORT" will never be sent. The actual call is "ROLLBACK" according to sql_parser.y